### PR TITLE
disttask: refine manager retry

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #49008 

Problem Summary:

### What changed and how does it work?
Use handle.RunWithRetry
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/disttask/framework/taskexecutor/manager.go [hunk 1]
```diff
@@ -367,17 +340,31 @@ func (m *Manager) logErr(err error) {
...
+	err1 := m.runWithRetry(func() error {
+		return m.taskTable.FailSubtask(m.ctx, m.id, taskID, err)
+	}, "update to subtask failed")
+	if err1 == nil {
+		m.logger.Error("update error to subtask success", zap.Int64("task-id", taskID), zap.Error(err1), zap.Stack("stack"))
+	}
+}
+
+func (m *Manager) runWithRetry(fn func() error, msg string) error {
+	backoffer := backoff.NewExponential(scheduler.RetrySQLInterval, 2, scheduler.RetrySQLMaxInterval)
```
Comment (zimulala) [inline pkg/disttask/framework/taskexecutor/manager.go:360]: Why do we not put this to `RunWithRetry`? Then we can use it directly.

File: pkg/disttask/framework/taskexecutor/manager.go [hunk 2]
```diff
@@ -367,17 +340,31 @@ func (m *Manager) logErr(err error) {
...
+	err1 := handle.RunWithRetry(m.ctx, scheduler.RetrySQLTimes, backoffer, m.logger,
+		func(ctx context.Context) (bool, error) {
+			return true, fn()
+		},
+	)
 	if err1 != nil {
-		m.logger.Error("update to subtask failed", zap.Error(err1), zap.Stack("stack"))
+		m.logger.Warn(msg, zap.Error(err1))
```
Comment (zimulala) [inline pkg/disttask/framework/taskexecutor/manager.go:367]: It seems that the error log is printed in `RunWithRetry`, so could we remove this log?

Comment (ywqzzy) [reply pkg/disttask/framework/taskexecutor/manager.go:367]: ok

File: pkg/disttask/framework/taskexecutor/manager.go (hunk 1 above)
Comment (ywqzzy) [reply pkg/disttask/framework/taskexecutor/manager.go:360]: maybe too many params?

**Review by D3Hunter — APPROVED** [review]

**Review by okJiang — APPROVED** [review]

**Pull request merged**